Gems/ROS2/Code/Source/RobotImporter/URDF/VisualsMaker.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include "RobotImporter/Utils/SourceAssetsStorage.h"
+#include "UrdfParser.h"
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/smart_ptr/shared_ptr.h>
+#include <RobotImporter/Utils/SourceAssetsStorage.h>
+
+namespace ROS2
+{
+    //! Populates a given entity with all the contents of the <visual> tag in robot description
+    class VisualsMaker
+    {
+    public:
+        VisualsMaker(
+            const std::map<std::string, urdf::MaterialSharedPtr>& materials,
```

Comment:
Can these parameter be map of `const AZStd::map<AZStd::string, urdf:MaterialSharedPtr>&`?

Reply:
It is done this way to have a clean way to construct this object: ` m_visualsMaker(model->materials_, urdfAssetsMapping)` in URDFPrefabMaker.  The model is of type `urdf::ModelInterfaceSharedPtr`. 
I rather left it as is, to have a conversion from std::map to AZStd::map inside this class.

Reply:
Yes, but lets hear from @lumberyard-employee-dm

Reply:
If the API to is interfacing with the URDF, then this is fine to remanding a std::map.

Reply:
Done